akka-docs/src/main/paradox/typed/interaction-patterns.md

```diff
@@ -10,44 +10,86 @@ The fundamental way to interact with an actor is through @scala["tell", which is
+In many interactions between actors a message leads to a response message back being sent back from the other actor, it can be the result of a query or some form of acknowledgment that the message was received and processed. In Akka Typed the recipient of responses has to be encoded as a field in the message itself, which the recipient can then use to send a response back. 
 
-TODO sample
+With the following protocol:
 
-**Useful when:**
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #request-response-protocol }
 
- * Subscribing to an actor that will send many response messages (of the same protocol) back
- * When communicating between a parent and its children, where the protocol can be made include the messages for the interaction 
- * ???
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-protocol }
+
+
+The sender would use its own @scala[`ActorRef[Response]`]@java[`ActorRef<Response>`], which it can access through @scala[`ActorContext.self`]@java[`ActorContext.getSelf()`], for the `respondTo`. 
+
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #request-response-send }
+
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-send }
+
+
+On the receiving side the `ActorRef` can then be used to send one or more responses back:
+
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #request-response-respond }
+
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-respond }
+
+
+**Useful when:**
 
+ * Subscribing to an actor that will send many response messages back
+ * Translating between different actor message protocols
+ 
 **Problems:**
 
- * Often the response that the other actor wants to send back is not a part of the sending actor's protocol (see adapted request response or ask)
- * It is hard to detect and that a message request was not delivered or processed (see ask)
- * Unless the protocol already includes a way to provide context, for example a request id that is also sent in the response, it is not possible to tie an interaction to some specific context without introducing a new, separate, actor
+ * Actors seldom have a response message from another actor as a part of their protocol (see adapted response)
```

Comment:
even though it's the immediate next section, perhaps make `adapted response` a link to make it more obvious it's a section?

Reply:
Yeah, that's my intention for all of those when done with the actual content 👍